nextcloudappstore/core/views.py

```diff
@@ -89,8 +91,46 @@ def get_context_data(self, **kwargs):
             context['current_category'] = Category.objects.get(id=category_id)
         if self.search_terms:
             context['search_query'] = ' '.join(self.search_terms)
+        context['get_arg_url_strings'] = self.get_arg_url_strings
         return context
 
     @cached_property
+    def get_arg_url_strings(self):
+        """URL encoded strings with the GET args of the last request.
+
+        Intended for preserving GET arguments upon clicking a link by including
+        one (and only one) of these strings in the "href" attribute.
+
+        The GET params are divided into three groups: search, filters and
+        ordering. In addition to these three, the returned dict also contains
+        some combinations of them, as specified by the dict keys.
+
+        No leading "?" or "&".
+
+        :return dict with URL encoded strings.
+        """
+
+        search = self._get_args_as_url_string('search')
+        filters = self._get_args_as_url_string('featured', 'maintainer')
+        ordering = self._get_args_as_url_string('order_by', 'ordering')
+
+        return {
+            'search': search,
+            'filters': filters,
+            'ordering': ordering,
+            'filters_ordering': self._combine_arg_strings(filters, ordering),
+            'search_filters': self._combine_arg_strings(search, filters),
+        }
+
+    def _get_args_as_url_string(self, *arg_names):
+        args = map(lambda arg: (arg, self.request.GET.get(arg, '')), arg_names)
+        filtered_dict = dict(filter(lambda a: a[1], args))
```

Comment:
may be more readable to move the filter out of the dict function and call the result present_args